tailnet/controllers.go

```diff
@@ -104,6 +104,7 @@ type WorkspaceUpdatesClient interface {
 
 type WorkspaceUpdatesController interface {
 	New(WorkspaceUpdatesClient) CloserWaiter
+	CurrentState() *proto.WorkspaceUpdate
```

Comment:
This should not be part of this interface, because the `Controller` doesn't care about the current state.  It's analogous to the methods we call on the `tailnet.Conn` --- those don't show up on any of the "controller" interfaces.

Instead, export the `TunnelAllWorkspaceUpdatesController` type.

Also think about what the right API is between the `vpn.Tunnel` and the controller, and consider creating new interface(s) that codify it, so that we can mock things out in testing.

Reply:
That makes sense, will do.
Though my inspiration here was the `ResumeTokenController` exposing a `Token` function, which does read from the current (last known) state of the internal `refresher`. Is this just an exception to the rule?